contrib/storage-hive/core/src/test/java/org/apache/drill/exec/hive/HiveTestFixture.java

```diff
@@ -0,0 +1,263 @@
+  }
+
+  public static Builder builder(File baseDir) {
+    return new Builder(requireNonNull(baseDir, "Parameter 'baseDir' can't be null!"));
+  }
+
+  public HivePluginManager getPluginManager() {
+    return pluginManager;
+  }
+
+  public HiveDriverManager getDriverManager() {
+    return driverManager;
+  }
+
+  public String getWarehouseDir() {
+    String warehouseDir = pluginConf.get(ConfVars.METASTOREWAREHOUSE.varname);
+    return nonNull(warehouseDir) ? warehouseDir : driverConf.get(ConfVars.METASTOREWAREHOUSE.varname);
+  }
+
+  public class HivePluginManager {
+
+    private HivePluginManager() {
+    }
+
+    public void addHivePluginTo(Drillbit... drillbits) {
+      addHivePluginTo(Arrays.asList(drillbits));
+    }
+
+    public void addHivePluginTo(Iterable<Drillbit> drillbits) {
+      try {
+        for (Drillbit drillbit : drillbits) {
+          HiveStoragePluginConfig pluginConfig = new HiveStoragePluginConfig(new HashMap<>(pluginConf));
+          pluginConfig.setEnabled(true);
+          drillbit.getContext().getStorage().createOrUpdate(pluginName, pluginConfig, true);
+        }
+      } catch (ExecutionSetupException e) {
+        ErrorHelper.sneakyThrow(e);
+      }
+    }
+
+    public void removeHivePluginFrom(Drillbit... drillbits) {
+      removeHivePluginFrom(Arrays.asList(drillbits));
+    }
+
+    public void removeHivePluginFrom(Iterable<Drillbit> drillbits) {
+      drillbits.forEach(bit -> bit.getContext().getStorage().deletePlugin(pluginName));
+    }
+
+    public void updateHivePlugin(Iterable<Drillbit> drillbits,
+                                 Map<String, String> configOverride) {
+      try {
+        for (Drillbit drillbit : drillbits) {
+          StoragePluginRegistry pluginRegistry = drillbit.getContext().getStorage();
+          HiveStoragePlugin storagePlugin = Objects.requireNonNull(
+              (HiveStoragePlugin) pluginRegistry.getPlugin(pluginName),
+              String.format("Hive storage plugin with name '%s' doesn't exist.", pluginName));
+
+          HiveStoragePluginConfig newPluginConfig = storagePlugin.getConfig();
+          newPluginConfig.getConfigProps().putAll(configOverride);
+          pluginRegistry.createOrUpdate(pluginName, newPluginConfig, true);
+        }
+      } catch (ExecutionSetupException e) {
+        ErrorHelper.sneakyThrow(e);
+      }
+    }
+
+  }
+
+  public class HiveDriverManager {
+
+    private HiveDriverManager() {
+    }
+
+    public void doWithinSession(Consumer<Driver> driverConsumer) {
+      final HiveConf hiveConf = new HiveConf(SessionState.class);
+      driverConf.forEach(hiveConf::set);
+      SessionState ss = new SessionState(hiveConf);
+      try (AutoCloseable toBeClosed = ss::close) {
+        SessionState.start(ss);
+        driverConsumer.accept(new Driver(hiveConf));
+      } catch (Exception e) {
+        ErrorHelper.sneakyThrow(e);
```

Comment:
I don't think this appropriate handling of sneaky throw. Its primary goal to help with exceptions within functional interfaces where we cannot re-throw them or handle them. In your case you can do one of them.

Reply:
Narrowed and re-thrown as runtime.